In [15]:
#import Relevant packages 
import pandas as pd

In [16]:
# load the Contents data sets
content = pd.read_csv("Content.csv")

# check the head of the data set
content.head()

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
2           2  230c4e4d-70c3-461d-b42c-ec09396efb3f   
3           3  356fff80-da4d-4785-9f43-bc1261031dc6   
4           4  01ab84dd-6364-4236-abbb-3f237db77180   

                                User ID   Type        Category  \
0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo        Studying   
1  beb1f34e-7870-46d6-9fc7-2e12eb83ce43  photo  healthy eating   
2  a5c65404-5894-4b87-82f2-d787cbee86b4  photo  healthy eating   
3  9fb4ce88-fac1-406c-8544-1a899cee7aaf  photo      technology   
4  e206e31b-5f85-4964-b6ea-d7ee5324def1  video            food   

                                                 URL  
0  https://socialbuzz.cdn.com/content/storage/975...  
1  https://socialbuzz.cdn.com/content/storage/9f7...  
2  https://socialbuzz.cdn.com/content/storage/230...  
3  https://socialbuzz.cdn.com/content/storage/356...  
4  https://socialbuzz.cdn.com/content/storage/01a...

Before we proceed with our Cleaning , we can see that our data contained 6 columns, some of which are irrelevant to our analysis below are the Explaination of each column; 

* Unnamed: 0 : This column should be dropped as it is the index that came with our file and python read it as a column and has ssigned a new index for our datas.

* Content ID: this is Useful in our Analysis as it Contained the unique ID for our content , and remember we are to Determine the Top 5 most Popular content, Hence, this Column should be kept.

* User ID : This column is irrelevant to our analysis it contain the ID for each user , but we are *NOT* to determine Top 5 most popular cateegory based on User, Hence this column should be drop.

* Type : This column contain The type of Contents we are go to do our Analysis on , so it should be Kept, But we need to change its name To "Content Type" To avoid Confusion.

* Category : This Column is Very useful for our Analysis as it's the category of each Content Type, Hence should be kept.


* URL : This Column is the link to social Site of The Company we are Conducting Analysis for and where the contents are stored , Nevertheless, It is Irrelevant for our Analysis and should be drop.

In [17]:
# Drop irrelevant Columns.

content.drop(["Unnamed: 0","User ID", "URL"], axis = 1, inplace = True)

# Change the the name of "Type" column to "Content Type"

content.rename(columns= {"Type":"Content Type"},inplace=True)

# Check the head of our file
content.head()


Content ID Content Type        Category
0  97522e57-d9ab-4bd6-97bf-c24d952602d2        photo        Studying
1  9f737e0a-3cdd-4d29-9d24-753f4e3be810        photo  healthy eating
2  230c4e4d-70c3-461d-b42c-ec09396efb3f        photo  healthy eating
3  356fff80-da4d-4785-9f43-bc1261031dc6        photo      technology
4  01ab84dd-6364-4236-abbb-3f237db77180        video            food

That's Cool, The next is to check for missing values in our data set and drop them if any .

In [18]:
# Check for missing values and arrange them in ascending other 
content.isna().sum().sort_values()

Content ID      0
Content Type    0
Category        0
dtype: int64

As we can see our content doesn"t have any missing Values. Cool right?, OK,

Let's Check "Content Type"  and "Category" Columns value counts 

In [21]:
# Check the value counts for Content Type 

content["Content Type"].value_counts()

photo    261
video    259
GIF      244
audio    236
Name: Content Type, dtype: int64

That's Nice "Content Type" Has four types, we have contents that are

* photo to be 261, 

* video 259 , 

* GIF 244 , and

* audio 236  and they are all well structured.

Let's proceed to "Category" Column

In [22]:
# check the value counts for Category column

content["Category"].value_counts()

technology           71
animals              67
travel               67
culture              63
science              63
fitness              61
food                 61
healthy eating       61
cooking              60
soccer               58
tennis               58
education            57
dogs                 56
studying             55
veganism             48
public speaking      48
Fitness               5
Animals               4
Science               4
"soccer"              3
"culture"             3
Soccer                3
"dogs"                2
Education             2
Studying              2
Travel                2
Food                  2
"veganism"            1
"public speaking"     1
Public Speaking       1
"technology"          1
"cooking"             1
Healthy Eating        1
"studying"            1
"food"                1
Culture               1
"tennis"              1
Technology            1
"animals"             1
Veganism              1
"science"             1
Name: Category, 

Woow!, This is whole lot, we can see the "Category" column isn't well Structured and it needs the folowing cleanings to be perfom on it:

* Removing all punctuations 

* ensuring all categories start with the same case(lower or upper). i.e removing capitalization.

All this can be done at a go with .apply() method and lambda function

In [23]:
# Clean the Category column 
content['Category'] = content['Category'].apply(lambda x:x.strip('"').lower())

# Check the value counts of the Category column again
content["Category"].value_counts()

technology         73
animals            72
travel             69
science            68
culture            67
fitness            66
food               64
soccer             64
healthy eating     62
cooking            61
tennis             59
education          59
studying           58
dogs               58
public speaking    50
veganism           50
Name: Category, dtype: int64

Looks, like our Content file is cleaned 

Let's save it as a csv file 



In [24]:
# save the content file as csv

content.to_csv("Cleaned Content.csv")